share/sysbuild/images/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2021-2023 Nordic Semiconductor
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# The primary application is the first image to be added to the build, so that
+# it is available while processing the remaining images.
+ExternalZephyrProject_Add(
+  APPLICATION ${DEFAULT_IMAGE}
+  SOURCE_DIR ${APP_DIR}
+  APP_TYPE MAIN
+)
+
+# This allows for MCUboot to be included.
+sysbuild_add_subdirectory(bootloader)
+
+# Include zephyr modules generated sysbuild CMake file.
+foreach(SYSBUILD_CURRENT_MODULE_NAME ${SYSBUILD_MODULE_NAMES})
+  # Note the second, binary_dir parameter requires the added
+  # subdirectory to have its own, local cmake target(s). If not then
+  # this binary_dir is created but stays empty. Object files land in
+  # the main binary dir instead.
+  # https://cmake.org/pipermail/cmake/2019-June/069547.html
+  zephyr_string(SANITIZE TOUPPER MODULE_NAME_UPPER ${SYSBUILD_CURRENT_MODULE_NAME})
+  if(NOT ${SYSBUILD_${MODULE_NAME_UPPER}_CMAKE_DIR} STREQUAL "")
+    set(SYSBUILD_CURRENT_MODULE_DIR ${SYSBUILD_${MODULE_NAME_UPPER}_MODULE_DIR})
+    set(SYSBUILD_CURRENT_CMAKE_DIR ${SYSBUILD_${MODULE_NAME_UPPER}_CMAKE_DIR})
+    sysbuild_add_subdirectory(${SYSBUILD_CURRENT_CMAKE_DIR}
+                              ${CMAKE_BINARY_DIR}/modules/${SYSBUILD_CURRENT_MODULE_NAME})
+  endif()
+endforeach()
```

Comment:
I understand why this is moved here, as today's main benefit of Sysbuild modules is to include extra images to the build.

But as sysbuild evolves in future, then a Sysbuild module may do more / others things than image adding, in which case having this processing placed under `images` may seem unlogical.

I do believe we should be careful to move this under the `images` folder.

Reply:
Well, the only reason this is moved here is for the processing order. But you have a good point about placing this under `images` and what this might communicate. I don't mean to imply that the sole purpose of sysbuild modules (or `sysbuild.cmake` files for that matter) is to add more images.

I see this as a naming issue. The directory doesn't need to be named `images`, but I don't know what else to call it. It could be more along the lines of `main`, `root`, or `core`, but those don't sit right with me.

That directory only exists to be the outermost one included with `sysbuild_add_subdirectory()` - other than that, it's completely artificial. Once it's renamed, I think the sysbuild modules should stay here.

Reply:
> That directory only exists to be the outermost one included with sysbuild_add_subdirectory()

if that's the reason then we could also consider to just create `app` and `boards` folders with a CMakeLists.txt file in each, and then at the top-level `sysbuild/CMakeLists.txt` have:
```
# sysbuild/CMakeLists.txt
...
sysbuild_add_subdirectory(app)
sysbuild_add_subdirectory(bootloader)

# sysbuild modules loop
...
foreach(SYSBUILD_CURRENT_MODULE_NAME ${SYSBUILD_MODULE_NAMES})
  ...
  sysbuild_add_subdirectory(${SYSBUILD_CURRENT_CMAKE_DIR} ...)
endforeach()
...

sysbuild_add_subdirectory(boards)
```

and then remove the `images` folder.

But regardless of whether we do that, or find a better name for the `images` folder, then that is a simple refactoring and can be done after this PR has been merged.

Thanks for sharing the intentions behind the `images` folder.
As we are clear on the purpose of the folder and the naming issues, then let's get this PR, as the functionality is more important than the small refactoring, which can always be done.